### PR TITLE
Vector table relay

### PR DESCRIPTION
**Pull request by oyvindronningstad**
subsys: bootloader: Use Zephyr's SW_VECTOR_RELAY functionality for nRF51
    
Implemented in arch/arm/core/irq_relay.S.
    
The linker changes define sections and symbols needed by irq_relay.S.
Some sections need to be placed at the very beginning of flash/RAM,
so they are defined before #include<nrf_common.ld>.

Together with https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/302 and https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/303 this makes the immutable bootloader work on nRF51.

### CODE REVIEW

File: subsys/bootloader/bootloader.c [hunk 1]
```diff
@@ -85,6 +85,16 @@ void uninit_used_peripherals(void)
 #endif
 }
 
+#ifdef CONFIG_SW_VECTOR_RELAY
+#ifndef CONFIG_SB_C_RUNTIME_SETUP_VARIANT_ZEPHYR
+_GENERIC_SECTION(.vt_pointer_section) u32_t _vector_table_pointer;
+#endif
+extern u32_t _vector_table_pointer;
+#define _VTOR _vector_table_pointer
```
Comment (lemrey) [inline subsys/bootloader/bootloader.c]: Why not simply `VTOR`? It's confusing to prefix symbols with underscores.. are they linker generated? are they compiler intrinsicts? are they "private"?

Comment (oyvindronningstad) [reply subsys/bootloader/bootloader.c]: Good point, I'll change it. This was left from a previous iteration.

Comment (oyvindronningstad) [reply subsys/bootloader/bootloader.c]: Done

**Comment by oyvindronningstad** [comment]
The PR was broken after publishing. I opened a new PR at #335

**Pull request closed without merging**